R/collect.R

```diff
@@ -358,11 +358,25 @@ collector <- function(x, coll_col = ".predictions") {
   } else {
     keep_cols <- coll_col
   }
-  x <- dplyr::select(x, dplyr::starts_with("id"), !!!keep_cols)
-  x <- tidyr::unnest(x, cols = c(dplyr::all_of(coll_col)))
+
+  id_cols <- colnames(x)[grepl("id", colnames(x))]
+  keep_cols <- c(coll_col, id_cols)
+  x <- x[, keep_cols]
+
+  res <-
+    vctrs::vec_cbind(
+      vctrs::list_unchop(x[[coll_col]]),
```

Comment:
As used in tune, `coll_col` is always one of `.metrics` or `.predictions`--i.e. always has length one.